### PR TITLE
Migrate codec-http2 to junit5

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should update to use junit5 in all modules.

Modifications:

Adjust codec-http2 tests to use junit5

Result:

Part of https://github.com/netty/netty/issues/10757

### CODE REVIEW

File: codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java [hunk 1]
```diff
@@ -366,16 +374,18 @@ public void emptyDataFrameShouldApplyFlowControl() throws Exception {
...
-        } finally {
-            data.release();
-        }
+        assertThrows(Http2Exception.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                decode().onDataRead(ctx, STREAM_ID, data, 10, true);
+            }
+        });
+        data.release();
```
Comment (chrisvest) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java:388]: Note that if the assertion fails, this pattern will leak. Not sure how much we care about buffer leaks in _failing_ tests, though.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java [hunk 2]
```diff
@@ -198,68 +221,102 @@ public void testLiteralHuffmanEncodedWithEmptyNameAndValue() throws Http2Excepti
...
-    @Test(expected = Http2Exception.class)
+    @Test
     public void testIncompleteIndex() throws Http2Exception {
         byte[] compressed = StringUtil.decodeHexDump("FFF0");
-        ByteBuf in = Unpooled.wrappedBuffer(compressed);
+        final ByteBuf in = Unpooled.wrappedBuffer(compressed);
         try {
-            hpackDecoder.decode(0, in, mockHeaders, true);
-            assertEquals(1, in.readableBytes());
-            hpackDecoder.decode(0, in, mockHeaders, true);
```
Comment (chrisvest) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java:241]: I wonder if the second call was the one that was supposed to throw, but the old test clearly couldn't tell the difference.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/HpackTest.java [hunk 3]
```diff
@@ -67,8 +57,9 @@ public HpackTest(String fileName) {
         return data;
     }
 
-    @Test
-    public void test() throws Exception {
+    @ParameterizedTest(name = "file = {0}")
+    @MethodSource("data")
+    public void test(String fileName) throws Exception {
```
Comment (chrisvest) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/HpackTest.java:62]: I think it's no longer necessary for `data` to return the arguments as a collection of _arrays_. It should be possible to just be a collection/stream/list/array of strings. For instance, the `File[]` array could be returned directly if the test took `File` as its argument.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java [hunk 4]
```diff
@@ -114,9 +110,12 @@ public void run() {
...
 
-            exceptionRule.expect(ExecutionException.class);
-            exceptionRule.expectCause(IsInstanceOf.<Throwable>instanceOf(ClosedChannelException.class));
-            promise.get(3, SECONDS);
+            try {
+                promise.get(3, SECONDS);
+                fail();
+            } catch (ExecutionException e) {
+                assertThat(e.getCause(), IsInstanceOf.<Throwable>instanceOf(ClosedChannelException.class));
+            }
```
Comment (chrisvest) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java:118]: If you use `assertThrows`, it'll return the exception thrown for further checks.

**Review by chrisvest — APPROVED** [review]
Looks good. Couple of comments, but nothing important.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java (hunk 1 above)
Comment (normanmaurer) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java:388]: yeah... I think it is fine either way.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java (hunk 2 above)
Comment (normanmaurer) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java:241]: when I worked on this the first one was actually throwing... Thats why I rewrote it.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java (hunk 4 above)
Comment (normanmaurer) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java:118]: TIL... Let me refactor

File: codec-http2/src/test/java/io/netty/handler/codec/http2/HpackTest.java (hunk 3 above)
Comment (normanmaurer) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/HpackTest.java:62]: TIL... Let me refactor

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java (hunk 4 above)
Comment (hyperxpro) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java:118]: Out of context question, What does `TIL` mean?

Comment (chrisvest) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrapTest.java:118]: @hyperxpro Today I Learned

**Review by chrisvest — APPROVED** [review]

**Pull request merged**